yacs/config.py

```diff
@@ -227,17 +248,55 @@ def raise_key_rename_error(self, full_key):
 
 
 def load_cfg(cfg_file_or_string):
-    """Load a cfg from a file or string."""
+    """Load a cfg. Supports loading from:
+        - A file object backed by a YAML file
+        - A file object backed by a Python source file that exports an attribute
+          "cfg" that is either a dict or a CfgNode
+        - A string that can be parsed as valid YAML
+    """
     _assert_with_logging(
         isinstance(cfg_file_or_string, _FILE_TYPES + (str,)),
         "Expected first argument to be of type {} or {}, but it was {}".format(
             _FILE_TYPES, str, type(cfg_file_or_string)
         ),
     )
+    if isinstance(cfg_file_or_string, str):
+        cfg_as_dict = yaml.safe_load(cfg_file_or_string)
+        return CfgNode(cfg_as_dict)
     if isinstance(cfg_file_or_string, _FILE_TYPES):
-        cfg_file_or_string = "".join(cfg_file_or_string.readlines())
-    cfg_as_dict = yaml.safe_load(cfg_file_or_string)
-    return _to_cfg_node(cfg_as_dict)
+        YAML_EXTS = {"", ".yaml", ".yml"}
+        PY_EXTS = {".py"}
```

Comment:
nit: should we make `YAML_EXTS` and `PY_EXTS` into module constants?

Reply:
Done as well.